src/core/mapping/instance_manager.h

```diff
@@ -152,8 +152,7 @@ class InstanceManager {
   void erase(Instance inst);
 
  public:
-  void lock();
-  void unlock();
+  Legion::Mapping::LocalLock& lock() { return manager_lock_; }
```

Comment:
I wouldn't name this `lock()`, as then `instance_manager->lock()` implies that a lock is being taken (and could lead to a bug in the future). `get_lock()` or something similar would be better.

Reply:
yeah, `lock()` seems ambiguous as it's unclear whether it's holding the lock or simply returning it. I renamed it with `manager_lock()`. FYI, I reserved `get_*` names for getters that involve some computation and return fresh values and not references.